### PR TITLE
FEAT: Manual XP grants

### LINKED ISSUE #826: [FEAT] - Manual Experience Point Grants
### What feature would you like to see?

Grant server managers the ability to award and remove experience.

This should be a *sub-command* under the existing `manage` top-level command. The sub-command should be called `xpmodify`, and should:

- Take a `user` option for the user to modify. Name this `user`.
- Take a `number` (NOT integer) option for the amount to adjust the XP by. Name this `adjustment`.
- The helper module should assert that the caller has the `MANAGE_SERVER` permission
- The helper module should assert that the adjustment would neither reduce the user's XP below 0 or increase it above the level cap.
- If all checks pass, the helper module should adjust the user's XP by the `adjustment` amount and save it.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Vyvy-vi**
# Pull Request

## Description

This PR add the `xpmodify` subcommand for the `manage` command, which allows a user with `MANAGE_GUILD` permissions to modify the XP of the server members.

> /manage xpmodify action: add/remove user: discord.GuildMember adjustment: int

## Screenshots

Screenshot 2021-12-16 at 7 20 49 PM
Screenshot 2021-12-16 at 7 20 01 PM

## Related Issue

Closes #826 

## Documentation

For _any_ version updates, please verify if the [documentation page](https://www.beccalyria.com/discord-documentation) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [ ] My contribution does NOT require a documentation update.
- [x] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/config/BeccaResponses.ts [hunk 1]
```diff
@@ -38,6 +38,14 @@ export const BeccaPhrases: BeccaLyria["responses"] = {
...
     "I punish you on my terms, not yours. And I don't feel like doing so today. <:BeccaYawn:867102883823353937>",
   ],
+  noSelfXP: [
+    "I try not to judge people's requests, but giving XP to yourself seems like a loser move <:BeccaPeek:867102878588862514>",
+    "I can not let you do that <:BeccaSmug:872577688022421566>",
+  ],
+  noBotXP: [
+    "Bots don't level up. They're already too poweful for you to handle <:BeccaWink:867102883635789844>",
+    "Thanks, but we bots don't need any XP coming from you... <:BeccaYikes:877278299066347632>",
+  ],
```
Comment (nhcarrigan) [inline src/config/BeccaResponses.ts:48]:
```suggestion
  noSelfXP: [
    "I try not to judge people's requests, but giving XP to yourself seems like a loser move. <:BeccaPeek:867102878588862514>",
    "I can not let you do that. <:BeccaSmug:872577688022421566>",
  ],
  noBotXP: [
    "Bots don't level up. They're already too powerful for you to handle. <:BeccaWink:867102883635789844>",
    "Thanks, but we bots don't need any XP coming from you... <:BeccaYikes:877278299066347632>",
  ],
```
I love these 😆

File: src/modules/commands/subcommands/manage/handleXpModify.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+import levelScale from "../../../../config/listeners/levelScale";
+import { LevelOptOut } from "../../../../config/optout/LevelOptOut";
+import LevelModel from "../../../../database/models/LevelModel";
+import { CommandHandler } from "../../../../interfaces/commands/CommandHandler";
+import { beccaErrorHandler } from "../../../../utils/beccaErrorHandler";
+import { getRandomValue } from "../../../../utils/getRandomValue";
+import { errorEmbedGenerator } from "../../../commands/errorEmbedGenerator";
+
+/**i
```
Comment (nhcarrigan) [inline src/modules/commands/subcommands/manage/handleXpModify.ts:12]:
```suggestion
/**
```

**Review by nhcarrigan — CHANGES_REQUESTED** [review]

File: src/modules/commands/subcommands/manage/handleXpModify.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+ * Allows the suggestion embed with the given `id` to be marked as approved or
+ * denied (determined by the `action`). Appends the `action` and the `reason` to the
+ * suggestion embed.
```
Comment (nhcarrigan) [inline src/modules/commands/subcommands/manage/handleXpModify.ts:15]: Also this JSDoc comment needs updated.

File: src/config/BeccaResponses.ts (hunk 1 above)
Comment (Vyvy-vi) [reply src/config/BeccaResponses.ts:48]: :D

**Comment by nhcarrigan** [comment]
## Thank You

Thank you for your continued contribution to our project. We have reviewed your pull request and are happy to accept these changes.

Please continue to watch for issues labelled `help wanted`, as these will be additional opportunities to contribute.

**Comment by Vyvy-vi** [comment]
Sorry for taking a bit long 😅

**Pull request merged**